lib/src/client/dart_http.dart

```diff
@@ -10,12 +13,42 @@ class DartHttp implements HttpHandler {
     final response = await _send(Request(request.method, request.uri)
       ..headers.addAll(request.headers)
       ..body = request.body);
-    return HttpResponse(response.statusCode,
-        body: response.body, headers: response.headers);
+    final responseBody =
+        _encodingForHeaders(response.headers).decode(response.bodyBytes);
```

Comment:
Could you add a test for this change please?

Reply:
Done